spec/immutable_struct_spec.rb

```diff
@@ -103,14 +103,28 @@ def self.from_array(array)
   end
 
   describe "to_h" do
-    it "should include the output of params and block methods in the hash" do
+    it "should not include the output of params and block methods in the hash" do
       klass = ImmutableStruct.new(:flappy) do
         def lawsuit
           'pending'
         end
       end
       instance = klass.new(flappy: 'bird')
-      instance.to_h.should == {flappy: 'bird', lawsuit: 'pending'}
+      instance.to_h.should == {flappy: 'bird'}
     end
+
+    it "allows #== to be overriden" do
```

Comment:
Yes, it could be changed to that. The issue happened when I subclassed an ImmutableStruct I'm using as an entity class.